Dockerfile

```diff
@@ -58,13 +64,18 @@ ENV GITEA_CUSTOM /data/gitea
 
 VOLUME ["/data"]
 
-ENTRYPOINT ["/usr/bin/entrypoint"]
+ENTRYPOINT ["/usr/bin/entrypoint.sh"]
 CMD ["/bin/s6-svscan", "/etc/s6"]
 
 COPY docker/root /
 COPY --from=build-env /go/src/code.gitea.io/gitea/gitea /app/gitea/gitea
 COPY --from=build-env /go/src/code.gitea.io/gitea/environment-to-ini /usr/local/bin/environment-to-ini
 COPY --from=build-env /go/src/code.gitea.io/gitea/contrib/autocompletion/bash_autocomplete /etc/profile.d/gitea_bash_autocomplete.sh
-RUN chmod 755 /usr/bin/entrypoint /app/gitea/gitea /usr/local/bin/gitea /usr/local/bin/environment-to-ini
-RUN chmod 755 /etc/s6/gitea/* /etc/s6/openssh/* /etc/s6/.s6-svscan/*
+RUN chmod 755 /usr/bin/entrypoint.sh \
+              /app/gitea/gitea \
+              /usr/local/bin/gitea \
+              /usr/local/bin/environment-to-ini \
+              /etc/s6/gitea/* \
+              /etc/s6/openssh/* \
+              /etc/s6/.s6-svscan/*
```

Comment:
IIRC there is a bug in the `chmod` step.

For example:

* "latest" 108M:  https://hub.docker.com/layers/gitea/gitea/latest/images/sha256-c77a7cf73ad59dd09c3ada400bbf585d12a772e9867dc574d3cfdaa76f92bbcf?context=explore
* "1.21-nightly" 65M: https://hub.docker.com/layers/gitea/gitea/1.21-nightly/images/sha256-d0b0c740cb6b9f0a0d381199dd2084d6a4a27c5a528750e0d80a043a3b43ec04?context=explore

The problem is the `RUN /bin/sh -c chmod 755` layer introduces another 44M change in "latest" image.

I think we should do chmod in the `build-env` stage, then the `chmod` layer won't cause so much a big change.

Reply:
Moved `chmod` to build-env.